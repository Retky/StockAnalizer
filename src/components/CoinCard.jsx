import React from 'react';
import PropTypes from 'prop-types';

const CoinCard = (props) => {
  const {
    id,
    symbol,
    price,
  } = props;

  return (
    <div className="card">
      <h2>{symbol}</h2>
      <small>{id}</small>
      <p>
        $
        {price}
      </p>
    </div>
  );
};

CoinCard.propTypes = {
  id: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
};

export default CoinCard;
