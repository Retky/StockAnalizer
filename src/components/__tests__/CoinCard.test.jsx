import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/configureStore';
import CoinCard from '../CoinCard';

const coin = {
  changePercent24Hr: '-2.8770796633988240',
  id: 'bitcoin',
  priceUsd: '29067.8651115798164805',
  symbol: 'BTC',
};

describe('CoinCard component', () => {
  test('component match with the snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <CoinCard
            key={coin.id}
            id={coin.id}
            symbol={coin.symbol}
            price={coin.priceUsd.substring(0, 8)}
            change={coin.changePercent24Hr}
          />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
