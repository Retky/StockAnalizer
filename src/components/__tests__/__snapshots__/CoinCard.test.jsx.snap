// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinCard component component match with the snapshot 1`] = `
<a
  className="card"
  href="/detail"
  onClick={[Function]}
>
  <h2>
    BTC
  </h2>
  <small>
    bitcoin
  </small>
  <p>
    $
    29067.86
  </p>
</a>
`;
